### PR TITLE
log obi commands to systemd journal on remote hosts

### PR DESCRIPTION
**Pull request by Gobiner**
Better than PR https://github.com/Oblong/obi/pull/94, this PR logs obi metadata and shell commands executed to systemd's journal.

### CODE REVIEW

**Comment by EchoAbstract** [comment]
While this checks for the systemd command before running it, so I think it should be safe, we do have users that have iMacs as their obi target systems.

**Comment by Gobiner** [comment]
```
┌[arice@aricembp] [/dev/ttys005] [255]
└[~]> ssh -t wh-hex-c journalctl -r --utc -t obi
Hint: You are currently not seeing messages from other users and the system.
      Users in groups 'adm', 'systemd-journal' can see all messages.
      Pass -q to turn off this notice.
-- Logs begin at Wed 2016-12-14 22:07:37 GMT, end at Wed 2017-03-15 17:41:34 GMT. --
Mar 15 17:41:34 hex-c-1 obi[1134]: obi build by arice: cmake --build "/tmp/arice/sea/build" -- -j32 -l32
Mar 15 17:41:29 hex-c-1 obi[673]: obi build by arice: cmake -H"/tmp/arice/sea" -B"/tmp/arice/sea/build" -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_EXPORT_COMPILE_COMMANDS=1 -DSEA_DCHECK_IS_ON=1
Mar 15 17:41:29 hex-c-1 obi[668]: obi build by arice: mkdir -p /tmp/arice/sea/build
Mar 15 17:41:28 hex-c-1 obi[605]: obi build by arice: fabric rsync local=/Users/arice/sea/ remote=/tmp/arice/sea
Mar 15 17:41:28 hex-c-1 obi[601]: obi build by arice: mkdir -p /tmp/arice/sea
Mar 15 17:41:22 hex-c-1 obi[568]: obi clean by arice: rm -rf /tmp/arice/sea/build || true
Mar 15 01:33:59 hex-c-1 obi[8977]: obi clean by arice: rm -rf /tmp/arice/sea/build || true
Mar 15 01:33:20 hex-c-1 obi[8863]: obi build by arice: cmake --build "/tmp/arice/sea/build" -- -j32 -l32
Mar 15 01:33:20 hex-c-1 obi[8845]: obi build by arice: cmake -H"/tmp/arice/sea" -B"/tmp/arice/sea/build" -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_EXPORT_COMPILE_COMMANDS=1 -DSEA_DCHECK_IS_ON=1
Mar 15 01:33:19 hex-c-1 obi[8840]: obi build by arice: mkdir -p /tmp/arice/sea/build
Mar 15 01:33:18 hex-c-1 obi[8777]: obi build by arice: fabric rsync local=/Users/arice/sea/ remote=/tmp/arice/sea
Mar 15 01:33:17 hex-c-1 obi[8773]: obi build by arice: mkdir -p /tmp/arice/sea
Mar 15 01:33:01 hex-c-1 obi[8711]: obi rsync by arice: fabric rsync local=/Users/arice/sea/ remote=/tmp/arice/sea
Mar 15 01:33:00 hex-c-1 obi[8707]: obi rsync by arice: mkdir -p /tmp/arice/sea
Mar 15 01:32:52 hex-c-1 obi[8673]: obi stop by arice: pkill -SIGTERM -f '[a-z/]+sea .*' || true
Connection to wh-hex-c closed.
┌[arice@aricembp] [/dev/ttys005]
└[~]>
```

**Comment by jshrake** [comment]
I'm against this MR because it introduces a systemd dependency for a feature that I don't think anyone would actually use.

I'll open the floor for the rest of the team to override my opinion.

**Comment by Gobiner** [comment]
It doesn't add a dependency on systemd: if `journalctl` is available on a remote host, obi will log to it. If not, then noop, no problem.

**Comment by Gobiner** [comment]
> fabric.operations.run(*args, **kwargs)
> Run a shell command on a remote host.
> 
> If shell is True (the default), run will execute the given command string via a shell interpreter, the value of which may be controlled by setting env.shell (defaulting to something similar to /bin/bash -l -c "<command>".)

File: obi/task/task.py [hunk 1]
```diff
@@ -309,3 +330,23 @@ def project_yaml():
...
+    if not s: return "''"
+    if _find_unsafe(s) is None: return s
+
+    # use single quotes, and put single quotes into double quotes
+    # the string $'b is then quoted as '$'"'"'b'
+    return "'" + s.replace("'", "'\"'\"'") + "'"
+
+def write_to_journal(string_input):
+    """
+    Run systemd-cat echo 'obi go by arice: message' on remote hosts
```
Comment (tomjakubowski) [inline obi/task/task.py]: this is weird, mixing the example in with the documentation itself. Just describe what's logged and give an example later.

**Comment by jshrake** [comment]
Closing per https://github.com/Oblong/obi/pull/104#issuecomment-293321004

**Pull request closed without merging**